### PR TITLE
PHYVersion API clarifications

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

References https://github.com/TheThingsNetwork/lorawan-stack/issues/3513

#### Changes

- Rename the fields of the `PHYVersion` enum based on the details enumerated here https://github.com/TheThingsNetwork/lorawan-stack/issues/2784#issuecomment-648735723 . The renamed enum values are backwards compatible in their JSON representation and binary representation.
- Add the RP2 `PHYVersion`s.
- Clarify in `pkg/band` which RP/TS versions are used for different versions of the band.
- Changes usages of the old naming scheme to the new naming scheme.
- Rename `MinAckTimeout` and `MaxAckTimeout` to `MinRetransmitTimeout` and `MaxRetransmitTimeout`, as this is the terminology used by RP2.

#### Testing

Unit testing.

##### Regressions

There should be none, as most of the work is just automated renaming.

#### Notes for Reviewers

This PR is meant to be the initial work for https://github.com/TheThingsNetwork/lorawan-stack/issues/3513 . It allows us to have correct names for the regional parameters without breaking compatibility.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/devicerepository/store/remote/enums.go
Comment (adriansmares) [inline pkg/devicerepository/store/remote/enums.go]: This list has been re-ordered based on the TS001/RP001/RP002 releases.

File: pkg/networkserver/networkserver_flow_test.go
Comment (adriansmares) [inline pkg/networkserver/networkserver_flow_test.go:63]: I've removed the ordering of versions, as `RP002` has clashing definitions for `1.0.0` and `1.0.1`.

File: pkg/ttnpb/testdata/ttnpb_encoding_golden.md
Comment (adriansmares) [inline pkg/ttnpb/testdata/ttnpb_encoding_golden.md:601]: Note that the old representations are unchanged.

File: pkg/ttnpb/testdata/ttnpb_encoding_golden.md
Comment (adriansmares) [inline pkg/ttnpb/testdata/ttnpb_encoding_golden.md:1193]: Note that the old representations are unchanged.

**Review by htdvisser — APPROVED** [review]

File: api/lorawan.proto
Comment (johanstokking) [inline api/lorawan.proto]: This one is actually Rev A

**Review by johanstokking — COMMENTED** [review]
I think you should take a shower after merging this PR

**Review by KrishnaIyer — APPROVED** [review]
May god be with you.

**Comment by johanstokking** [comment]
@adriansmares are you holding the merge for the decisions and implications regarding PHY? Or what is the reason to hold?

**Comment by adriansmares** [comment]
> 
> 
> @adriansmares are you holding the merge for the decisions and implications regarding PHY? Or what is the reason to hold?

It needs approval from a code owner - you and/or @neoaggelos .

**Review by johanstokking — APPROVED** [review]

**Review by neoaggelos — APPROVED** [review]
Also tested backwards compatibility with previous enum values (e.g. `PHY_V1_0_2_REV_B`) to make sure that nothing breaks. :+1:

**Pull request merged**
